fs/fat/fs_fat32.c

```diff
@@ -764,7 +764,13 @@ static ssize_t fat_write(FAR struct file *filep, FAR const char *buffer,
 
   /* Check if the file size would exceed the range of off_t */
 
-  if (ff->ff_size + buflen < ff->ff_size)
+#if defined(CONFIG_FS_LARGEFILE)
+#  define OFF_MAX INT64_MAX
+#else
+#  define OFF_MAX INT32_MAX
+#endif
```

Comment:
Let's add "Pre-processor definitions" comment section and place defines there. Please see an example in other C-files.